Nix CI failed since irb 1.8.0 required rdoc gem as a dependency

https://github.com/ruby/irb/pull/648
https://github.com/kachick/irb-power_assert/actions/runs/6032136518/job/16366855692

```
An error occurred while installing psych (5.1.0), and Bundler cannot continue.

In Gemfile:
  debug was resolved to 1.8.0, which depends on
    irb was resolved to 1.8.0, which depends on
      rdoc was resolved to 6.5.0, which depends on
        psych
Error: Process completed with exit code 5.
```

--- a/flake.nix
+++ b/flake.nix
@@ -29,6 +29,11 @@
               bashInteractive
 
               ruby
+              # Required to build psych via irb dependency
+              # https://github.com/kachick/irb-power_assert/issues/116
+              # https://github.com/ruby/irb/pull/648
+              libyaml
+
               tree
               nil
               nixpkgs-fmt

